frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/switch-to-form-validation-utils.ts

```diff
@@ -0,0 +1,94 @@
+import i18n from 'i18next';
+import * as yup from 'yup';
+import { nameValidationSchema } from '@console/shared';
+import { PipelineTask } from '../../../types';
+
+const resourceDefinitionLite = () => {
+  return yup.array().of(
+    yup.object({
+      name: yup.string(),
+      resource: yup.string(),
+    }),
+  );
+};
+
+const taskValidationSchemaLite = () => {
+  return yup.array().of(
+    yup
+      .object({
+        name: nameValidationSchema((tKey) => i18n.t(tKey)),
+        taskRef: yup
+          .object({
+            name: yup.string(),
+            kind: yup.string(),
+          })
+          .default(undefined),
+        taskSpec: yup.object(),
+        runAfter: yup.array().of(yup.string()),
+        params: yup.array().of(
+          yup.object({
+            name: yup.string(),
+            value: yup.lazy((value) => {
+              if (Array.isArray(value)) {
+                return yup.array().of(yup.string());
+              }
+              return yup.string();
+            }),
+          }),
+        ),
+        resources: yup.object({
+          inputs: resourceDefinitionLite(),
+          outputs: resourceDefinitionLite(),
+        }),
+        when: yup.array().of(
+          yup.object({
+            input: yup.string(),
+            operator: yup.string(),
+            values: yup.array().of(yup.string()),
+          }),
+        ),
+
+        workspaces: yup.array().of(
+          yup.object({
+            name: yup.string(),
+            workspace: yup.string(),
+          }),
+        ),
+      })
+      .test(
+        'taskRef-or-taskSpec',
+        i18n.t('pipelines-plugin~TaskSpec or TaskRef must be provided.'),
+        function(task: PipelineTask) {
+          return !!task.taskRef || !!task.taskSpec;
+        },
+      ),
+  );
+};
+
+export const pipelineBuilderFormSchemaLite = () => {
+  return yup.object({
+    metadata: yup.object({ name: yup.string() }),
```

Comment:
```suggestion
    metadata: yup.object({ name: nameValidationSchema(...) }),
```

Interestingly enough, when editing this can't change either. Since we are updating the resource by that name.